src/security/ldap/src/main/java/org/geoserver/security/ldap/BindingLdapAuthoritiesPopulator.java

```diff
@@ -146,6 +160,8 @@ public BindingLdapAuthoritiesPopulator(ContextSource contextSource, String group
         if (password != null) {
             // authenticate and execute role extraction in the authenticated
             // context
+            Supplier<DirContext> ctxSupplier =
```

Comment:
Is this needed? Why don't you use the authenticate ctx from ldapTemplate.authenticate anymore?

Reply:
Indeed, using authenticate is the right way.  This was used because reuse Context objects is not allowed after doing a search.  Fixing it, thanks.

Reply:
Refactored to use a functional consumer that wraps authentication process and propagate to its called methods.

Reply:
ok, for the future I would prefer if you don't force push, so that I can see in a new commit the changes

Reply:
Ok, makes sense.